syslog/syslog.go

```diff
@@ -143,10 +143,13 @@ func (l *Logger) writePacket(p Packet) {
 			l.connect()
 		}
 
+		deadline := time.Now().Add(time.Duration(30) * time.Second)
```

Comment:
Instead of hardcoding this, I think we should add a `timeout` field to `Logger` so it can be passed in and configured by the configuration file.